src/access/acp.ts

```diff
@@ -236,3 +251,69 @@ function ruleAppliesTo(
 ): boolean {
   return rule !== null && getIriAll(rule, actorRelation).includes(actor);
 }
+
+/**
+ * Get a set of all actors mentionned in an ACR by active Rules (i.e. that are
+ * references by Policies referenced by the ACR Control, and therefore that
+ * effectively apply).
+ *
+ * @param resource The resource with the ACR we want to inspect
+ * @param actorRelation
+ */
+function internal_findActorAll(
+  resource: WithAccessibleAcr & WithResourceInfo,
+  actorRelation: typeof acp.agent | typeof acp.group
+): Set<WebId> {
+  const actors: Set<WebId> = new Set();
+  // Collect all policies that apply to the resource or its ACR (aka active)
+  const activePolicyUrls = getPolicyUrlAll(resource).concat(
+    getAcrPolicyUrlAll(resource)
+  );
+  const rules = getActiveRuleAll(resource, activePolicyUrls);
+  // This code could be prettier using flat(), which isn't supported by nodeJS 10.
+  // If you read this comment after April 2021, feel free to refactor.
+  rules.forEach((ruleUrl) => {
+    // The rules URL being extracted from the dataset, it is safe to assume
+    // that getThing cannot return undefined.
+    const ruleThing = getThing(internal_getAcr(resource), ruleUrl)!;
+    getUrlAll(ruleThing, actorRelation)
+      .filter(
+        (iri) =>
+          !([acp.PublicAgent, acp.CreatorAgent] as string[]).includes(iri) || actorRelation != acp.agent
+      )
+      .forEach((iri) => actors.add(iri));
+  });
+  return actors;
+}
+
+/**
+ * Iterate through all the actors active for an ACR, and list all of their access.
+ * @param resource The resource for which we want to list the access
+ * @param actorRelation The type of actor we want to list access for
+ * @returns A map with each actor access indexed by their WebID, or null if some
```

Comment:
Since groups don't have a WebID:

```suggestion
 * @returns A map with each actor access indexed by their identifying URL, or null if some
```